routes/api/users.js

```diff
@@ -27,20 +30,57 @@ async (req,res) => {
 
     try{
         let user = await User.findOne({ email });
+        
+        //See if user exists
 
-        if(user){
-            res.status(400).json({errors: [{msg: 'User already exists'}]});
+         if(user){
+           return res.status(400).json({errors: [{msg: 'User already exists'}]});
         }
         
-        //See if user exists
         //Get user gravatar
+
+        const avatar = gravatar.url(email, {
+            s: '200',
+            r: 'pg',
+            d: 'mm'
+        })
+
+        user = new User({
+            name, 
+            email,
+            avatar,
+            password
+        })
+
         //Encrypt Password
+
+        const salt = await bcrypt.genSalt(10);
+
+        user.password = await bcrypt.hash(password, salt);
```

Comment:
can you just open a node cli on terminal and give an example of what these values will look like just to get better understanding of what bcrypt is doing?
1) type node on terminal
2) now you are in node shell so just :
    const bcrypt = require('bcryptjs');        
    const salt = await bcrypt.genSalt(10);
    and print salt and then generate password and print password?